### PR TITLE
Add pre-commit config

### PR DESCRIPTION
**Pull request by excitedleigh**
Does what it says on the tin.

Note that you'll need to tag releases with the version number on Git to have the instructions in the readme actually work though!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,6 +30,19 @@ or in single-file mode.
...
 6. Iterates those steps until the source code stops changing.
 
+## Using with pre-commit
+
+If you use [pre-commit](https://pre-commit.com/), you can use it with Shed by
+adding the following to your `.pre-commit-config.yaml`:
+
+```yaml
+repos:
+- repo: https://github.com/Zac-HD/shed.git
```
Comment (hugovk) [inline README.md:40]: No need for `.git`:
```suggestion
- repo: https://github.com/Zac-HD/shed
```

Compare https://github.com/asottile/pyupgrade#as-a-pre-commit-hook

**Comment by Zac-HD** [comment]
Hi all - I'm delighted that you're interested in `shed` :smile: 

I'm just holding off on promoting it more widely until `isort` 5.0 is released, because with 4.x it's impossible for `shed` to configure it cleanly.  Once that's done though, pre-commit integration sound pretty cool to me!

**Comment by Zac-HD** [comment]
Notes (mostly to myself):  I definitely want to support `pre-commit`, even if I and my students don't use it :smile: 

- [ ] https://pre-commit.com/#filtering-files-with-types instead of a `files: ...` pattern
- [ ] Single-file mode *in the CLI* will need to grow first-party-module detection *if* in a git repo.
- [ ] Tags should be handled by GitHub actions - after uploading to PyPI, tag the branch with current version.
- [ ] We currently have tests that the changelog and `__version__` attribute are in sync.  We should make the changelog single source of truth, and rewrite `__version__` and the new pre-commit config and docs to match from a script in `tox -e check`.

**Comment by Zac-HD** [comment]
Closed in d66b6f4c1858494065634fd8193c01082fa8c4be.

Thanks again for the PR @excitedleigh, and sorry it took so long to get to it!

**Pull request closed without merging**